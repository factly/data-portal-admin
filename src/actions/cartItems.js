import axios from '../utils/axios';
import {
  ADD_CARTITEMS,
  SET_CARTITEM_LOADING,
  SET_CARTITEM_REQUEST,
  SET_CARTITEM_IDS,
  CARTITEM_API,
} from '../constants/cartItems';
import { addProducts } from './products';
import { getIds, getValues, deleteKeys, buildObjectOfItems } from '../utils/objects';

export const loadCartItems = (id, page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      cartItems: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setCartItemIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${CARTITEM_API(id)}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { id, page: page, limit: limit, ids: currentPageIds };
    dispatch(setCartItemRequest(currentReq, total));
    dispatch(addCartItems(nodes));
    dispatch(setCartItemIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

const setLoading = (loading) => {
  return {
    type: SET_CARTITEM_LOADING,
    payload: { loading },
  };
};

export const addCartItems = (cartItems) => (dispatch) => {
  const products = getValues(cartItems, 'product');
  dispatch(addProducts(products));

  dispatch({
    type: ADD_CARTITEMS,
    payload: {
      cartItems: buildObjectOfItems(deleteKeys(cartItems, ['product'])),
    },
  });
};

const setCartItemRequest = (req, total) => {
  return {
    type: SET_CARTITEM_REQUEST,
    payload: { req, total },
  };
};

const setCartItemIds = (ids) => {
  return {
    type: SET_CARTITEM_IDS,
    payload: { ids },
  };
};
