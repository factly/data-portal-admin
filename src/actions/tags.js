import axios from '../utils/axios';
import {
  ADD_TAG,
  ADD_TAGS,
  SET_TAG_LOADING,
  SET_TAG_REQUEST,
  SET_TAG_IDS,
  RESET_TAG,
  TAG_API,
} from '../constants/tags';
import { getIds, buildObjectOfItems } from '../utils/objects';

export const loadTags = (page = 1, limit = 5) => {
  return async (dispatch, getState) => {
    const {
      tags: { req },
    } = getState();

    let ids;
    for (let item of req) {
      if (item.page === page && item.limit === limit) {
        ids = [...item.ids];
      }
    }

    if (ids) {
      dispatch(setTagIds(ids));
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${TAG_API}?page=${page}&limit=${limit}`,
      method: 'get',
    });

    const { nodes, total } = response.data;
    const currentPageIds = getIds(nodes);
    const currentReq = { page: page, limit: limit, ids: currentPageIds };
    dispatch(setTagRequest(currentReq, total));
    dispatch(addTags(nodes));
    dispatch(setTagIds(currentPageIds));

    dispatch(setLoading(false));
  };
};

export const createTag = (data) => {
  return async (dispatch, getState) => {
    dispatch(setLoading(true));

    await axios({
      url: TAG_API,
      method: 'post',
      data: data,
    });

    dispatch(resetTag());
    dispatch(setLoading(false));
  };
};

export const updateTag = (id, data) => {
  return async (dispatch, getState) => {
    let url = `${TAG_API}/${id}`;

    dispatch(setLoading(true));

    const response = await axios({
      url: url,
      method: 'put',
      data: data,
    });

    dispatch(addTag(response.data));
    dispatch(setLoading(false));
  };
};

export const deleteTag = (id) => {
  return async (dispatch, getState) => {
    let url = `${TAG_API}/${id}`;

    dispatch(setLoading(true));

    await axios({
      url: url,
      method: 'delete',
    });

    dispatch(resetTag());
    dispatch(setLoading(false));
  };
};

export const getTag = (id) => {
  return async (dispatch, getState) => {
    const {
      tags: { items },
    } = getState();

    if (items[id]) {
      return;
    }

    dispatch(setLoading(true));

    const response = await axios({
      url: `${TAG_API}/${id}`,
      method: 'get',
    });
    dispatch(addTag(response.data));

    dispatch(setLoading(false));
  };
};

const setLoading = (loading) => {
  return {
    type: SET_TAG_LOADING,
    payload: { loading },
  };
};

const addTag = (tag) => {
  return {
    type: ADD_TAG,
    payload: { tag },
  };
};

export const addTags = (tags) => {
  return {
    type: ADD_TAGS,
    payload: {
      tags: buildObjectOfItems(tags),
    },
  };
};

const setTagRequest = (req, total) => {
  return {
    type: SET_TAG_REQUEST,
    payload: { req, total },
  };
};

const setTagIds = (ids) => {
  return {
    type: SET_TAG_IDS,
    payload: { ids },
  };
};

const resetTag = () => {
  return {
    type: RESET_TAG,
  };
};
