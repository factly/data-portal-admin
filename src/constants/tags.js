export const LOADING_TAGS = 'LOADING_TAGS';
export const LOAD_TAGS_SUCCESS = 'LOAD_TAGS_SUCCESS';
export const SET_TAGS_LIST_TOTAL = 'SET_TAGS_LIST_TOTAL';
export const ADD_TAGS_LIST_REQUEST = 'ADD_TAGS_LIST_REQUEST';
export const SET_TAGS_LIST_CURRENT_PAGE = 'SET_TAGS_LIST_CURRENT_PAGE';
export const LOAD_TAGS_FAILURE = 'LOAD_TAGS_FAILURE';
export const CREATING_TAG = 'CREATING_TAG';
export const CREATE_TAG_SUCCESS = 'CREATE_TAG_SUCCESS';
export const CREATE_TAG_FAILURE = 'CREATE_TAG_FAILURE';
export const UPDATING_TAG = 'UPDATING_TAG';
export const UPDATE_TAG_SUCCESS = 'UPDATE_TAG_SUCCESS';
export const UPDATE_TAG_FAILURE = 'UPDATE_TAG_FAILURE';
export const DELETING_TAG = 'DELETING_TAG';
export const DELETE_TAG_SUCCESS = 'DELETE_TAG_SUCCESS';
export const DELETE_TAG_FAILURE = 'DELETE_TAG_FAILURE';

export const baseUrl = `${process.env.REACT_APP_API_URL}/tags`;
