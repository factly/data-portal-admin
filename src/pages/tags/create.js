import React from 'react';
import { Form, Input, Button, notification } from 'antd';

const formItemLayout = {
  labelCol: { span: 6 },
  wrapperCol: { span: 14 },
};

const TagCreate = (props) => {
  const onFinish = (values) => {
    fetch(process.env.REACT_APP_API_URL + '/tags', {
      method: 'POST',
      body: JSON.stringify(values),
    })
      .then((res) => {
        if (res.status === 201) {
          return res.json();
        } else {
          throw new Error(res.status);
        }
      })
      .then((_) => {
        notification.success({
          message: 'Success',
          description: 'Tag succesfully added',
        });
        props.history.push(`${process.env.PUBLIC_URL}/tags`);
      })
      .catch((res) => {
        notification.error({
          message: 'Error',
          description: 'Something went wrong',
        });
      });
  };

  return (
    <Form name="tags_create" {...formItemLayout} onFinish={onFinish}>
      <Form.Item
        label="Name"
        name="title"
        rules={[
          {
            required: true,
            message: 'Please enter name!',
          },
        ]}
      >
        <Input placeholder="Ex. Crime In India" />
      </Form.Item>

      <Form.Item
        label="Slug"
        name="slug"
        rules={[
          {
            required: true,
            message: 'Please enter slug!',
          },
        ]}
      >
        <Input placeholder="Ex. crime-in-india" />
      </Form.Item>

      <Form.Item wrapperCol={{ span: 12, offset: 6 }}>
        <Button type="primary" htmlType="submit">
          Submit
        </Button>
      </Form.Item>
    </Form>
  );
};

export default TagCreate;
