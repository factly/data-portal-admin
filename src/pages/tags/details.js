import React from "react";
import {
  useParams
} from "react-router-dom";

const TagDetail = () => {
  const { id } = useParams()
  return <h1>Tag Details {id}</h1>;
};

export default TagDetail;
