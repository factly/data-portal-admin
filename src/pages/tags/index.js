import React, { useState } from 'react';
import { Table, Input, Button, Popconfirm, Form, notification } from 'antd';
import moment from 'moment';
import { Link } from 'react-router-dom';
import { EditOutlined, DeleteOutlined, SaveOutlined, CloseOutlined } from '@ant-design/icons';

const EditableCell = ({ editing, dataIndex, title, record, index, children, ...restProps }) => {
  return (
    <td {...restProps}>
      {editing ? (
        <Form.Item
          name={dataIndex}
          style={{
            margin: 0,
          }}
          rules={[
            {
              required: true,
              message: `Please enter ${title}!`,
            },
          ]}
        >
          <Input />
        </Form.Item>
      ) : (
        children
      )}
    </td>
  );
};

const Tags = () => {
  const [form] = Form.useForm();
  const [data, setData] = useState([]);
  const [total, setTotal] = useState(0);
  const [editingKey, setEditingKey] = useState('');

  React.useEffect(() => {
    fetch(process.env.REACT_APP_API_URL + '/tags')
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  }, []);

  const get = (page, limit) => {
    cancel();
    fetch(process.env.REACT_APP_API_URL + '/tags?page=' + page + '&limit=' + limit)
      .then((data) => data.json())
      .then((data) => {
        setData(data.nodes);
        setTotal(data.total);
      });
  };

  const isEditing = (record) => record.id === editingKey;

  const edit = (record) => {
    form.setFieldsValue(record);
    setEditingKey(record.id);
  };

  const cancel = () => {
    setEditingKey('');
  };

  const save = async (key) => {
    try {
      const row = await form.validateFields();
      const index = data.findIndex((item) => item.id === key);

      if (index > -1) {
        const item = data[index];
        fetch(process.env.REACT_APP_API_URL + '/tags/' + item.id, {
          method: 'PUT',
          body: JSON.stringify(row),
        })
          .then((res) => {
            if (res.status === 200) {
              return res.json();
            } else {
              throw new Error(res.status);
            }
          })
          .then((res) => {
            const newData = [...data];
            newData.splice(index, 1, res);
            setData(newData);
            setEditingKey('');
            notification.success({
              message: 'Success',
              description: 'Tag succesfully updated',
            });
          })
          .catch(() => {
            notification.error({
              message: 'Error',
              description: 'Something went wrong',
            });
          });
      }
    } catch (err) {
      notification.warning({
        message: 'Warning',
        description: 'Validation failed',
      });
    }
  };

  const deleteTag = (key) => {
    const index = data.findIndex((item) => item.id === key);
    if (index > -1) {
      fetch(process.env.REACT_APP_API_URL + '/tags/' + key, {
        method: 'DELETE',
      })
        .then((res) => {
          if (res.status === 200) {
            const newData = [...data];
            newData.splice(index, 1);
            setData(newData);
            notification.success({
              message: 'Success',
              description: 'Tag succesfully deleted',
            });
          }
        })
        .catch((err) => {
          notification.error({
            message: 'Error',
            description: 'Something went wrong',
          });
        });
    }
  };

  const columns = [
    {
      title: 'Title',
      dataIndex: 'title',
      width: '25%',
      editable: true,
    },
    {
      title: 'Slug',
      dataIndex: 'slug',
      width: '25%',
      editable: true,
    },
    {
      title: 'Created At',
      dataIndex: 'created_at',
      width: '25%',
      render: (_, record) => {
        return <span title={record.created_at}>{moment(record.created_at).fromNow()}</span>;
      },
    },
    {
      title: 'Operation',
      dataIndex: 'operation',
      render: (_, record) => {
        const editable = isEditing(record);
        return editable ? (
          <span>
            <Button
              type="primary"
              icon={<SaveOutlined />}
              onClick={() => save(record.id)}
              style={{
                marginRight: 8,
              }}
            >
              Save
            </Button>
            <Popconfirm title="Sure to cancel?" onConfirm={cancel}>
              <Button icon={<CloseOutlined />}>Cancel</Button>
            </Popconfirm>
          </span>
        ) : (
          <span>
            <Button
              type="primary"
              icon={<EditOutlined />}
              disabled={editingKey !== ''}
              onClick={() => edit(record)}
              style={{
                marginRight: 8,
              }}
            >
              Edit
            </Button>
            <Popconfirm
              disabled={editingKey !== ''}
              title="Sure to delete?"
              onConfirm={() => deleteTag(record.id)}
            >
              <Button disabled={editingKey !== ''} icon={<DeleteOutlined />}>
                Delete
              </Button>
            </Popconfirm>
          </span>
        );
      },
    },
  ];

  const mergedColumns = columns.map((col) => {
    if (!col.editable) {
      return col;
    }

    return {
      ...col,
      onCell: (record) => ({
        record,
        dataIndex: col.dataIndex,
        title: col.title,
        editing: isEditing(record),
      }),
    };
  });

  return (
    <div>
      <Link to={'/tags/create'}>
        <Button type="primary" style={{ marginBottom: 16 }}>
          Add Tag
        </Button>
      </Link>
      <Form form={form} component={false}>
        <Table
          components={{
            body: {
              cell: EditableCell,
            },
          }}
          bordered
          rowKey="id"
          dataSource={data}
          columns={mergedColumns}
          rowClassName="editable-row"
          pagination={{
            defaultPageSize: 5,
            onChange: get,
            total: total,
          }}
        />
      </Form>
    </div>
  );
};

export default Tags;
