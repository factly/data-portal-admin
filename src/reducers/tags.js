import produce from 'immer';
import {
  ADD_TAG,
  ADD_TAGS,
  SET_TAG_LOADING,
  SET_TAG_REQUEST,
  SET_TAG_IDS,
  RESET_TAG,
} from '../constants/tags';

const initialState = {
  loading: false,
  ids: [],
  req: [],
  items: {},
  total: 0,
};

const tagsReducer = produce((draft, action = {}) => {
  switch (action.type) {
    case SET_TAG_LOADING:
      draft.loading = action.payload.loading;
      return;
    case ADD_TAG: {
      const { tag } = action.payload;
      draft.items[tag.id] = tag;
      return;
    }
    case ADD_TAGS: {
      const { tags } = action.payload;
      Object.assign(draft.items, tags);
      return;
    }
    case SET_TAG_IDS:
      draft.ids = action.payload.ids;
      return;
    case SET_TAG_REQUEST: {
      const { req, total } = action.payload;
      draft.req.push(req);
      draft.total = total;
      return;
    }
    case RESET_TAG:
      return initialState;
  }
}, initialState);

export default tagsReducer;
