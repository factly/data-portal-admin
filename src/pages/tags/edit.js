import React from 'react';
import { useParams, useHistory } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { Skeleton } from 'antd';

import TagCreateForm from './components/TagCreateForm';
import { getTag, updateTag } from '../../actions/tags';

const EditTag = () => {
  const { id } = useParams();
  const history = useHistory();
  const dispatch = useDispatch();
  const { tag } = useSelector(({ tags }) => tags);

  React.useEffect(() => {
    dispatch(getTag(id));
  }, []);

  if (!tag.id) return <Skeleton />;

  const onUpdate = async (values) => {
    await dispatch(updateTag(id, values));
    history.push('/tags');
  };

  return <TagCreateForm data={tag} onSubmit={onUpdate} />;
};

export default EditTag;
