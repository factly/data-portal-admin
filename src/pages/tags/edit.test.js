import React from 'react';
import { useHistory } from 'react-router-dom';
import renderer, { act as rendererAct } from 'react-test-renderer';
import { useDispatch, Provider, useSelector } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';

import '../../matchMedia.mock';
import EditTag from './edit';
import * as actions from '../../actions/tags';
import TagEditForm from './components/TagCreateForm';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
  useParams: jest.fn().mockReturnValue({ id: '1' }),
}));

jest.mock('../../actions/tags', () => ({
  getTag: jest.fn(),
  addTag: jest.fn(),
  updateTag: jest.fn(),
}));

describe('Tags List component', () => {
  let store = mockStore({
    tags: {
      req: [],
      items: {
        1: {
          id: 1,
          name: 'Tag-1',
          slug: 'tag-1',
          description: 'description',
        },
        2: {
          id: 2,
          name: 'Tag-2',
          slug: 'tag-2',
          description: 'description',
        },
      },
      loading: false,
    },
  });
  store.dispatch = jest.fn(() => ({}));

  const mockedDispatch = jest.fn();
  useDispatch.mockReturnValue(mockedDispatch);

  describe('snapshot testing', () => {
    it('should render the component', () => {
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditTag />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match component with empty data', () => {
      store = mockStore({
        tags: {
          req: [],
          items: {},
          loading: false,
        },
      });
      let component;
      store.details = {};
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditTag />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should match skeleton while loading', () => {
      store = mockStore({
        tags: {
          req: [],
          items: {},
          loading: true,
        },
      });
      let component;
      rendererAct(() => {
        component = renderer.create(
          <Provider store={store}>
            <EditTag />
          </Provider>,
        );
      });
      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper;
    beforeEach(() => {
      store = mockStore({
        tags: {
          req: [],
          items: {
            1: {
              id: 1,
              name: 'Tag-1',
              slug: 'tag-1',
              description: 'description',
            },
            2: {
              id: 2,
              name: 'Tag-2',
              slug: 'tag-2',
              description: 'description',
            },
          },
          loading: false,
        },
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should call get action', () => {
      actions.getTag.mockReset();
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditTag />
          </Provider>,
        );
      });
      expect(actions.getTag).toHaveBeenCalledWith('1');
    });
    it('should call updateTag', () => {
      actions.updateTag.mockReset();
      const push = jest.fn();
      useHistory.mockReturnValueOnce({ push });
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <EditTag />
          </Provider>,
        );
      });
      wrapper.find(TagEditForm).props().onSubmit({ test: 'test' });
      expect(actions.updateTag).toHaveBeenCalledWith('1', { test: 'test' });
      expect(push).toHaveBeenCalledWith('/tags');
    });
  });
});
