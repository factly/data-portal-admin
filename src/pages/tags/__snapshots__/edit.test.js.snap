// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags List component snapshot testing should match component with empty data 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="tags_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="tags_create_title"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="tags_create_title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Crime In India"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="tags_create_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="tags_create_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. crime-in-india"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;

exports[`Tags List component snapshot testing should match skeleton while loading 1`] = `
<div
  className="ant-skeleton"
>
  <div
    className="ant-skeleton-content"
  >
    <h3
      className="ant-skeleton-title"
      style={
        Object {
          "width": "38%",
        }
      }
    />
    <ul
      className="ant-skeleton-paragraph"
    >
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": undefined,
          }
        }
      />
      <li
        style={
          Object {
            "width": "61%",
          }
        }
      />
    </ul>
  </div>
</div>
`;

exports[`Tags List component snapshot testing should render the component 1`] = `
<form
  className="ant-form ant-form-horizontal"
  id="tags_create"
  onSubmit={[Function]}
>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="tags_create_title"
        title="Name"
      >
        Name
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="tags_create_title"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. Crime In India"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-6 ant-form-item-label"
      style={Object {}}
    >
      <label
        className="ant-form-item-required"
        htmlFor="tags_create_slug"
        title="Slug"
      >
        Slug
      </label>
    </div>
    <div
      className="ant-col ant-col-14 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <input
            className="ant-input"
            id="tags_create_slug"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            placeholder="Ex. crime-in-india"
            type="text"
            value="slug"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="ant-row ant-form-item"
    style={Object {}}
  >
    <div
      className="ant-col ant-col-12 ant-col-offset-6 ant-form-item-control"
      style={Object {}}
    >
      <div
        className="ant-form-item-control-input"
      >
        <div
          className="ant-form-item-control-input-content"
        >
          <button
            className="ant-btn ant-btn-primary"
            onClick={[Function]}
            type="submit"
          >
            <span>
              Submit
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
